scanner.c

```diff
@@ -38,34 +75,108 @@ scanner_clear(struct scanner_statistics * data) {
 	memset(data, 0, sizeof * data);
 }
 
+int get_next_field(char * buf, char ** line) {
+  int buf_i = 0;
+  // skip the previous tab
+  (*line)++;
+  while (**line != '\t') {
+    if (**line == '\0' || **line == '\n') return 0;
+    buf[buf_i] = **line;
+    (*line)++;
+    buf_i++;
+    if (buf_i == BUF_LEN) return 0;
+  }
+  buf[buf_i] = '/0';
```

Comment:
`/0` is incorrect it should be `\0`. This will store character `0` with ascii value 0x30 to `buf[buf_i]`.